File: src/main/java/no/nav/personopplysninger/features/endreopplysninger/domain/kontonummer/Kontonummer.java
package no.nav.personopplysninger.features.endreopplysninger.domain.kontonummer;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Kontonummer {
    private String kilde = "BRUKER SELV";
    private UtenlandskKontoInformasjon utenlandskKontoInformasjon;
    private String value;

    @JsonCreator
    public static Kontonummer create(String json) throws JsonParseException, JsonMappingException, IOException {
        return (new ObjectMapper()).readValue(json, Kontonummer.class);
    }

    public String getKilde() {
        return kilde;
    }

    public UtenlandskKontoInformasjon getUtenlandskKontoInformasjon() {
        return utenlandskKontoInformasjon;
    }

    public String getValue() {
        return value;
    }
}

File: src/main/kotlin/no/nav/personopplysninger/features/endreopplysningergammel/domain/kontonummer/Kontonummer.kt
package no.nav.personopplysninger.features.endreopplysningergammel.domain.kontonummer

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.core.JsonParseException
import com.fasterxml.jackson.databind.JsonMappingException
import com.fasterxml.jackson.databind.ObjectMapper

import java.io.IOException

@JsonIgnoreProperties(ignoreUnknown = true)
class Kontonummer {
    @JsonProperty("@type")
    private val subtype = "KONTONUMMER"

    val kilde = "BRUKER SELV"
    val utenlandskKontoInformasjon: UtenlandskKontoInformasjon? = null
    val value: String? = null

    companion object {

        @JsonCreator
        @JvmStatic
        @Throws(JsonParseException::class, JsonMappingException::class, IOException::class)
        fun create(json: String): Kontonummer {
            return ObjectMapper().readValue(json, Kontonummer::class.java)
        }
    }
}
